tests/jobs/test_local_task_job.py

```diff
@@ -438,5 +521,5 @@ def test_number_of_queries_single_loop(self, mock_get_task_runner, return_codes)
         mock_get_task_runner.return_value.return_code.side_effects = return_codes
 
         job = LocalTaskJob(task_instance=ti, executor=MockExecutor())
-        with assert_queries_count(12):
+        with assert_queries_count(13):
```

Comment:
added a new DB query in local_task_job to query task state after task finished execution so we can decide to which callback to invoke